### PR TITLE
Fix don't swallow the error when building zap logger

### PR DESCRIPTION
**Pull request by git-hulk**
**What changed?**
Fix don't swallow the error when building zap logger

**Why?**

Currently, start the cadence server would be panic when
failed to build the zap logger. It will cause confusion since
the error was swallowed.

**How did you test it?**

test local with no permission log dir

**Potential risks**
no

### CODE REVIEW

File: tools/cli/domainUtils.go [hunk 1]
```diff
@@ -299,7 +299,8 @@ func initializeDomainHandler(
 func initializeLogger(
 	serviceConfig *config.Config,
 ) log.Logger {
-	return loggerimpl.NewLogger(serviceConfig.Log.NewZapLogger())
+	zapLogger, _ := serviceConfig.Log.NewZapLogger()
```
Comment (meiliang86) [inline tools/cli/domainUtils.go:302]: log fatal or panic here in case of error?

**Review by meiliang86 — APPROVED** [review]

File: tools/cli/domainUtils.go (hunk 1 above)
Comment (meiliang86) [reply tools/cli/domainUtils.go:302]: I guess you don't have a logger, so the only thing you can do is to panic

Comment (git-hulk) [reply tools/cli/domainUtils.go]: Yeah, I have used the `ErrorAndExited` to log the error and exit

**Review by longquanzheng — APPROVED** [review]

File: bench/lib/context.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ type RuntimeContext struct {
 
 // NewRuntimeContext builds a runtime context from the config
 func NewRuntimeContext(cfg *Config) (*RuntimeContext, error) {
-	logger := cfg.Log.NewZapLogger()
+	logger, _ := cfg.Log.NewZapLogger()
```
Comment (longquanzheng) [inline bench/lib/context.go:57]: We should just panic here. having error in the result means the logger could be invalid.

Comment (git-hulk) [reply bench/lib/context.go]: fixed

**Comment by Groxx** [comment]
Huh.  Why don't we have this passed in?  Constructing a logger seems doomed to be over-complicated (as we expose more and more logger-configuration fields) and less flexible (zap loggers can do effectively anything, there's no reason for us to limit them).

Accepting a logger would also simplify the `log, _ := config.NewZapLogger()` case in tests: pass in a `zaptest.NewLogger(t)` instead of constructing one and trying to work around the test environment.

**Comment by longquanzheng** [comment]
> Huh. Why don't we have this passed in? Constructing a logger seems doomed to be over-complicated (as we expose more and more logger-configuration fields) and less flexible (zap loggers can do effectively anything, there's no reason for us to limit them).
> 
> Accepting a logger would also simplify the `log, _ := config.NewZapLogger()` case in tests: pass in a `zaptest.NewLogger(t)` instead of constructing one and trying to work around the test environment.

Which one are you referring to? I think we have to do this way because they are in from different `main()` : server, bench, canary and CLI. So we can't pass a logger from server to cli code, hehe. 
Or in other words, we already try the best to pass around/initiate only once.

**Comment by longquanzheng** [comment]
@Groxx I merged this PR but if you see a way that we can improve by passing around logger, we make do a refactoring

**Pull request merged**